wtransport/src/tls.rs

```diff
@@ -740,21 +780,22 @@ pub mod client {
     impl ServerCertVerifier for ServerHashVerification {
         fn verify_server_cert(
             &self,
-            end_entity: &rustls::Certificate,
-            _intermediates: &[rustls::Certificate],
-            _server_name: &rustls::ServerName,
-            _scts: &mut dyn Iterator<Item = &[u8]>,
+            end_entity: &rustls_pki_types::CertificateDer,
+            _intermediates: &[rustls_pki_types::CertificateDer],
+            _server_name: &rustls_pki_types::ServerName,
             _ocsp_response: &[u8],
-            now: std::time::SystemTime,
+            now: rustls_pki_types::UnixTime,
         ) -> Result<ServerCertVerified, rustls::Error> {
             use time::OffsetDateTime;
             use x509_parser::oid_registry::OID_EC_P256;
             use x509_parser::oid_registry::OID_KEY_TYPE_EC_PUBLIC_KEY;
             use x509_parser::time::ASN1Time;
 
-            let now = ASN1Time::new(OffsetDateTime::from(now));
+            let now = ASN1Time::new(
+                OffsetDateTime::from_unix_timestamp(now.as_secs() as i64).expect("time overflow"),
+            );
```

Comment:
It would have panicked before, inside `OffsetDateTime::from(system_time)`, because of `checked_add`.